sympy/solvers/ode/single.py

```diff
@@ -425,42 +432,40 @@ def _get_general_solution(self, *, simplify: bool = True):
         return [gensol]
 
 
-
 class AlmostLinear(SinglePatternODESolver):
     r"""
     Solves an almost-linear differential equation.
 
     The general form of an almost linear differential equation is
 
-    .. math:: f(x) g(y) y + k(x) l(y) + m(x) = 0
-                \text{where} l'(y) = g(y)\text{.}
+    .. math:: a(x) g'(f(y)) f'(y) + b(x) g(f(y)) + c(x) = 0
 
```

Comment:
I think that this should be 
```
    .. math:: a(x) g'(f(x)) f'(x) + b(x) g(f(x)) + c(x) = 0
```
(there is not `y`).

Also the docstring should explicitly say that `f(x)` is the unknown function that is to be solved for.

Reply:
> Also the docstring should explicitly say that `f(x)` is the unknown function that is to be solved for.

I think we solve the equation for `u(x)`. Also, `y` is there in the equation just to simplify the integral in the example. Otherwise, since a, b are functions of `x`, the integral would be returned as it is. Do you suggest I print the integral instead in the example doctest?

Reply:
We solve for u as an intermediate step. In the original ODE `f(x)` is the dependent variable and that's what the solution comes back as:
```
In [1]: dsolve(f(x)**2 * f(x).diff(x) + f(x)**3 + 1, f(x), hint='almost_linear')                                                               
Out[1]: 
⎡                                                ______________                        ______________⎤
⎢          ______________                     3 ╱     -3⋅x                          3 ╱     -3⋅x     ⎥
⎢       3 ╱     -3⋅x              (-1 - √3⋅ⅈ)⋅╲╱  C₁⋅ℯ     - 1          (-1 + √3⋅ⅈ)⋅╲╱  C₁⋅ℯ     - 1 ⎥
⎢f(x) = ╲╱  C₁⋅ℯ     - 1 , f(x) = ─────────────────────────────, f(x) = ─────────────────────────────⎥
⎣                                               2                                     2              ⎦
```

Reply:
Ohh I see. I'll update it accordingly.